### PR TITLE
Add accounts resource

### PR DESCRIPTION
**Pull request by griffinmyers**
This pull request adds support for fetching `accounts` and `transactions`.  

Because it involves a restructuring of the `Button::Response` class, It is not backwards-compatible and will require a major version bump with the next release.

### CODE REVIEW

File: lib/button/resources/resource.rb [hunk 1]
```diff
@@ -43,10 +44,13 @@ def timeout
...
     # @param [String] path the HTTP path
+    # @param [Hash=] query optional query params to send
     # @return [Button::Response] the API response
     #
-    def api_get(path)
-      api_request(Net::HTTP::Get.new(path))
+    def api_get(path, query = nil)
+      uri = URI(path)
+      uri.query = URI.encode_www_form(query) if query
+      api_request(Net::HTTP::Get.new(uri.to_s))
```
Comment (danielmcgrath) [inline lib/button/resources/resource.rb:53]: I don't believe you need the `to_s` here

**Review by danielmcgrath — APPROVED** [review]
Looks great!

File: lib/button/resources/resource.rb (hunk 1 above)
Comment (griffinmyers) [reply lib/button/resources/resource.rb:53]: I do, as far as I can tell.  Passing a `URI` causes a `Cannot duplicate: NilClass` error (I didn't check across MRI versions).  I dug around in rubydocs for awhile and didn't find anything (and in fact found snippets indicating passing `URI` is okay) describing the interface sufficiently that I could work things out.  I'll take another look today just to be sure.

Comment (griffinmyers) [reply lib/button/resources/resource.rb:53]: I *did* find a more sensible method than `#to_s` though: https://github.com/sj26/ruby-1.9.3-p0/blob/master/lib/net/http.rb#L260

Comment (danielmcgrath) [reply lib/button/resources/resource.rb:53]: Ah, this is because they're not full URLs and just paths. Carry on. :)

Comment (griffinmyers) [reply lib/button/resources/resource.rb:53]: Can't use `#request_uri` either, per:

```irb
2.3.0 :002 > URI('bloop.net').request_uri
NoMethodError: undefined method `request_uri' for #<URI::Generic bloop.net>
	from (irb):2
	from /Users/williammyers/.rvm/rubies/ruby-2.3.0/bin/irb:11:in `<main>'
2.3.0 :003 > URI('http://bloop.net').request_uri
 => "/"
```

**Comment by griffinmyers** [comment]
Okay, fire in the hole!  Thanks @danielmcgrath

**Pull request merged**